src/v/raft/consensus.cc

```diff
@@ -720,30 +718,37 @@ consensus::linearizable_barrier(model::timeout_clock::time_point deadline) {
         update_node_append_timestamp(target);
         vlog(
           _ctxlog.trace, "Sending empty append entries request to {}", target);
-        auto f = _client_protocol
-                   .append_entries(
-                     target.id(),
-                     std::move(req),
-                     rpc::client_opts(_replicate_append_timeout))
-                   .then([this, id = target.id(), seq, dirty_offset](
-                           result<append_entries_reply> reply) {
-                       process_append_entries_reply(
-                         id, reply, seq, dirty_offset);
-                   });
-
-        send_futures.push_back(std::move(f));
+
+        ssx::background
+          = ssx::spawn_with_gate_then(
+              _bg,
+              [this,
+               target,
+               req = std::move(req),
+               seq,
+               dirty_offset]() mutable {
+                  return _client_protocol
+                    .append_entries(
+                      target.id(),
+                      std::move(req),
+                      rpc::client_opts(_replicate_append_timeout))
+                    .then([this, id = target.id(), seq, dirty_offset](
+                            result<append_entries_reply> reply) {
+                        process_append_entries_reply(
+                          id, reply, seq, dirty_offset);
+                    });
+              })
+              .handle_exception(
+                [holder = _bg.hold(), this](const std::exception_ptr& e) {
```

Comment:
If this `.hold()` fails but the one in `ssx::spawn_with_gate_then` doesn't won't Seastar still complain on an abandoned exceptional future?